Balancer-Bots/Stable-Pool-Depegging/src/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { ethers } from "forta-agent";
+import BigNumber from "bignumber.js";
+import { NetworkManager } from "forta-agent-tools";
+import { AMP_UPDATE_STARTED_ABI } from "./constants";
+
+export interface NetworkData {
+  stablePoolAddresses: string[];
+  valueThreshold?: string;
+  decreaseThreshold?: string;
+  decreasePercentageThreshold?: string;
+}
+
+export const toBn = (ethersBn: ethers.BigNumber) => new BigNumber(ethersBn.toString());
+
+export type GetAmpUpdateStartedLog = (
+  blockNumber: number
+) => Promise<(ethers.utils.LogDescription & { emitter: string })[]>;
+
+export function provideGetAmpUpdateStartedLogs(
+  networkManager: NetworkManager<NetworkData>,
+  provider: ethers.providers.Provider
+): GetAmpUpdateStartedLog {
+  const stablePoolIface = new ethers.utils.Interface([AMP_UPDATE_STARTED_ABI]);
+  const topics = [stablePoolIface.getEventTopic("AmpUpdateStarted")];
+
+  return async (blockNumber: number) => {
+    const encodedLogs = await provider.getLogs({
+      topics,
+      fromBlock: blockNumber,
+      toBlock: blockNumber,
+    });
+
+    const stablePoolAddresses = networkManager.get("stablePoolAddresses");
+
+    return encodedLogs
+      .filter(
+        (log) => stablePoolAddresses.includes(log.address) || stablePoolAddresses.includes(log.address.toLowerCase())
```

Comment:
just one question: Is there a chance where the cases don't match? like `stablePoolAddresses` ['AAa0'] and the `log.address` is like 'Aaa0'. if so, maybe we should compare both as lower cases? if that's not possible then yeah, no problem

Reply:
Right, I understand what you're talking about. It is possible, while unusual, since if the address has uppercase characters it's a checksum address and two checksum addresses shouldn't be different probably.
But since it is possible, I'll be changing that.

Reply:
Changed at [88e5500](https://github.com/NethermindEth/Forta-Agents/pull/295/commits/88e550075f8ef31b4cb32d197d2cea971c006b49).